### PR TITLE
Make OSD screen bigger if possible

### PR DESCRIPTION
**Pull request by McGiverGim**
Fixes https://github.com/betaflight/betaflight-configurator/issues/2631

This is an attempt to make the OSD scale with large resolutions. This is the first versions, it needs:

- [x] To be cleaned
- [x] Tested under different devices resolutions.
- [x] Make some final adjustments

I publish it in this form to get some feedback and testing.

This is an example of before:
image

And after:
image

### CODE REVIEW

**Comment by Asizon** [comment]
@McGiverGim this is my result lol:

image

File: src/css/tabs/osd.css
Comment (Asizon) [inline src/css/tabs/osd.css:468]: should be osd-bg-1-grid-pal.png

**Review by Asizon — CHANGES_REQUESTED** [review]

**Comment by HThuren** [comment]
What about a possibility to change the OSD background  from the IDE ?

File: src/css/tabs/osd.css
Comment (McGiverGim) [reply src/css/tabs/osd.css:468]: Good catch! I will fix it later...

**Comment by McGiverGim** [comment]
> @McGiverGim this is my result lol:

@Asizon What screen resolution are you using? I think I forgot maybe give a min-size to the display. I will test it later.

> What about a possibility to change the OSD background from the IDE ?

@HThuren we have a petition of not do it :P

https://github.com/betaflight/betaflight-configurator/blob/6319421e3d00d1ec7e664247dc1221185c74b8c6/src/css/tabs/osd.css#L464-L466

But I think you can open a feature request for this. This petition is out of the scope of this PR.

**Comment by Asizon** [comment]
@McGiverGim not sure about my screen resolution, but is small, i can check it later.

**Comment by haslinghuis** [comment]
`1600x900`:

image

File: src/css/tabs/osd.css
Comment (haslinghuis) [inline src/css/tabs/osd.css:542]: Invalid property according to chrome devtools:

Screenshot from 2022-05-09 22-17-44

Changing `min-content` to `auto` fixes display on Ubuntu

Comment (haslinghuis) [reply src/css/tabs/osd.css:542]: Change still works on Windows (but chrome devtools does not complain here)

Comment (McGiverGim) [reply src/css/tabs/osd.css]: Modified directly to `flex: 2 0 360px` and removed the min-width. I think works better.

**Comment by McGiverGim** [comment]
Modified, rebased and cleaned. Ready for a real review and test.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by benlumley** [comment]
Screenshot 2022-05-10 at 10 19 06

Testing on MacOS here - there's a point here that may not be as intended (it's before the nav bar collapses)

One fix is to put flex-wrap: nowrap on the container (in an appropriate media query I guess?)

File: src/css/tabs/osd.css
Comment (haslinghuis) [reply src/css/tabs/osd.css]: Sorry but on Linux this does not work. Leaving out the last parameters works on Linux and also on WIndows on the 3 grid columns.
```
flex: 1 1
flex: 2 0
flex: 1 1
```

Comment (McGiverGim) [reply src/css/tabs/osd.css]: Maybe this is because the nw version is 0.51.1 (Linux) vs 0.60.0 (Windows)?

The third parameter is the `flex-basis`. If you don't specify it is like `auto`. Try to add an aditional parameter `flex-basis: 360px;` for example, to see if it accepts it.

**Comment by McGiverGim** [comment]
@benlumley thanks for the feedback...

Pushed a new version, I've adjusted the size of the columns to fit in the space just before the collapse of the menu. Is a little ugly, the texts and the selects of the OSD lateral columns wrap, but all remains in correct position. There is not too much that we can do with this small resolution.
Another solution is to move to the Android layout, only one colum with OSD first, but is better to limit this option to small screens.
The header goes wrong when you use the "virtual FC", because it takes space used by the header. Has nothing to do with this PR, but you can open a bug for it.

File: src/css/tabs/osd.css
Comment (haslinghuis) [reply src/css/tabs/osd.css]: Indeed flex-basis parameter is not accepted. Also tried with NW 0.62.0 on Ubuntu and it works.

**Review by haslinghuis — APPROVED** [review]
Can't test OSX otherwise it's LGTM

**Comment by mrRobot62** [comment]
Found a little hickup for this solution.

Min-Width of the window is limited as well as min-height of window but min-height limiting is to small, because OSD image is cropped. Min-height limitation should be a little bit bigger. Will include a screenshoot asap

**Comment by benlumley** [comment]
Just tested on mac...

Issue gone from me; looks good at all widths; here's a range of screenshots.... 

Screenshot 2022-05-11 at 09 43 12

Screenshot 2022-05-11 at 09 43 50

Screenshot 2022-05-11 at 09 43 43

Screenshot 2022-05-11 at 09 43 30

Screenshot 2022-05-11 at 09 43 22

Screenshot 2022-05-11 at 09 42 59

**Comment by mrRobot62** [comment]
on my Mac below images is min-width and min-height
image

Image is cropped

**Comment by McGiverGim** [comment]
@mrRobot62 what resolution is this? I will change it to take into account the height of the window.

**Comment by mrRobot62** [comment]
image

**Comment by McGiverGim** [comment]
A little magic in the CSS and now it adjusts to the height too... at the minimum height it is a little out of screen, but it was before this change too... the only solution is to limit a little more the minimum resolution.

File: src/css/tabs/osd.css
Comment (haslinghuis) [inline src/css/tabs/osd.css:544]:
```suggestion
  max-width: max(min(950px, calc((100vh - 470px) * 1.25)), 350px);
```

File: src/tabs/osd.html
Comment (haslinghuis) [inline src/tabs/osd.html:54]: 👾

File: src/css/tabs/osd.css
Comment (haslinghuis) [inline src/css/tabs/osd.css:549]: On Windows/Linux: 180px is enough.

File: src/css/tabs/osd.css
Comment (haslinghuis) [inline src/css/tabs/osd.css:537]: 180px seems enough on Windows / Linux

Comment (haslinghuis) [reply src/css/tabs/osd.css:537]: EDIT: But 300px seems needed to keep these two on the same line:

image

File: src/css/tabs/osd.css
Comment (McGiverGim) [reply src/css/tabs/osd.css:544]: Can you explain? This is not working? It does not adjust width (and height accordingly) when making the height of the window smaller?

File: src/tabs/osd.html
Comment (McGiverGim) [reply src/tabs/osd.html:54]: I suppose the correct indentation is missing. I will fix.

File: src/css/tabs/osd.css
Comment (McGiverGim) [reply src/css/tabs/osd.css:549]: You are thinking like an "English" user. There are tons of languages, so this:
- Is a reasonable width for any language. Some of them will be cut, but we can't do it better.
- Is a size that fits in the minimum width of the window without breaking the layout.

File: src/css/tabs/osd.css
Comment (McGiverGim) [reply src/css/tabs/osd.css:537]: The same than before. The other PR will make it better:

https://github.com/betaflight/betaflight-configurator/pull/2920

but under very low resolutions/window sizes, something will not be perfect. The size is limited.

File: src/css/tabs/osd.css
Comment (haslinghuis) [reply src/css/tabs/osd.css:544]: Indeed with smaller height the effect is noticed.

**Comment by McGiverGim** [comment]
Pushed new version with the little fixes of latest review.

**Review by haslinghuis — DISMISSED** [review]

**Comment by haslinghuis** [comment]
@McGiverGim please rebase

**Review by Asizon — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Comment by ctzsnooze** [comment]
Works well on my Mac!

**Review by ctzsnooze — APPROVED** [review]

**Comment by mrRobot62** [comment]
sorry guys - same issue as I postet days ago - MacBook Pro, Catalina

image

Monitor resolution
image

**Comment by mrRobot62** [comment]
my recommendation limit width and height auf the main window - like this

image

**Comment by McGiverGim** [comment]
Use the scroll to center the OSD. Of course limiting the size is another option but it's not necessary at this moment.

**Pull request merged**
